### DISCUSSION TITLE
mega menu > Content Width (There is no option for website width)

### DISCUSSION THREAD

**Discussion by Michaelgimii** [elementor/elementor#21811] [Feature Request]
### Prerequisites

- [X] I have searched for similar features requests in both open and closed tickets and cannot find a duplicate.
- [X] The feature is still missing in the latest stable version of Elementor ( Elementor Pro )

### What problem is your feature request going to solve? Please describe.

Hi @avivu👋
Since you manage Pro versions, I wanted to share an important topic with you.

The Mega Manu widget is really great, but I think it's missing an important option. In this widget, the Content Width option has two main options. Full width and fit to content. **But if we want the width of the megamenu panel to be equal to the width of the site (site settings > layout > content width), there is no option for it!** This is a very important option that does not exist.

### Describe the solution you'd like

Please add a new option to Content Width in the mega menu widget. If we choose this option, the width of the megamenu panel will be equal to the width of the website content.

![menu](https://user-images.githubusercontent.com/88034902/228514895-d68e2df2-ac24-48c8-a022-7c2328f87a14.png)

thanks

**Comment by nicholaszein** [comment]
Hello @Michaelgimii,

we've discussed this in the beta [here](https://github.com/orgs/elementor/discussions/21553#discussioncomment-5215669), and I showed you an example [here](https://github.com/orgs/elementor/discussions/21553#discussioncomment-5266370), but I believe this deserves a better explanation.

To achieve what you want, all you have to do is to set Menu's content width to Full Width, and leave the container as boxed (this is why it's the default value). Then you can add an inner container, and add a background color to it. This will make it span the default value for boxed containers and allow you to leave the menu as boxed.

Here's a video explaining what was done to achieve this layout:

https://user-images.githubusercontent.com/113993460/228536045-8208c278-8880-4441-8f17-3c31aea7d18a.mp4



### Steps to reproduce:

1. Add a **Menu** widget;
2. Set one of the menu items with `Dropdown Content`;
3. Leave the **Menu's Content Width** as `Full Width`;
4. Go to the Menu's **Style** tab, and in the **Content** section, make sure to set the `Background Color` to transparent;
5. In the canvas, select the menu item with dropdown content to reveal the container inside;
6. Leave the menu item container's **Content Width** as `Boxed`;
7. Drag and Drop a new **Container** inside the menu item's container;
8. In the new inner container's **Style** tab, set a `Background Color`;
9. If you wish, also set a drop-shadow.

You now have your Mega Menu with boxed content.

### Why did we choose to make it this way?

This aligns with the overall behavior of Containers and allows users to have the flexibility they need to build what they want without being limited to a single setting in the menu. You can do virtually any layout with containers. Nested Elements like the new Menu widget are meant to leverage this flexibility.

We have no intention to change this behavior at this moment because it would limit what people can do with the new Menu widget. So, for now, I am closing this Feature Request.

PS: Please, avoid tagging Elementor staff. We already see all the issues and discussions posted. Thanks

**Comment by Michaelgimii** [comment]
Hi @nicholaszein, I apologize for the delay.

I really appreciate the time you took to make the video. You are one person, but you manage GitHub Elementor in the best possible way.

I understand your solution. But there is one main problem. **The problem is that if we set the "Open On" option on the hover in the Megamenu widget, if the user moves the mouse in the designated areas as shown in the image below, the Megamenu content will not be hidden.**

![mega menu](https://github.com/elementor/elementor/assets/88034902/a3f46644-4510-428e-beec-58f5a016d29b)

In this case, users will definitely face problems. Am I wrong?
